lib/include/ert/ecl/ecl_util.h

```diff
@@ -118,8 +118,9 @@ bool            ecl_util_unified_file(const char *filename);
 const char    * ecl_util_file_type_name( ecl_file_enum file_type );
 char          * ecl_util_alloc_base_guess(const char *);
 int             ecl_util_filename_report_nr(const char *);
-ecl_file_enum   ecl_util_get_file_type(const char * , bool * , int * );
 ecl_file_enum   ecl_util_inspect_extension(const char * ext , bool *_fmt_file, int * _report_nr);
+ecl_file_enum   ecl_util_get_file_type__(const char * , bool * , int * , bool * valid_case);
```

Comment:
I thought double underscore meant "private" or "secret"?  Why use it here, and why expose it in the header?

Reply:
I must say that I find this API very peculiar.
```c
ecl_util_get_file_type(const char * , bool * , int * );
```
What are the necessary parameters here?

Reply:
> I thought double underscore meant "private" or "secret"? Why use it here, and why expose it in the header?

That is loosely what I imply here - or ~alternatively "the implementation" of the `ecl_util_get_file_type()` function - which is probably a more correct description in this case. It is used by the Python code, which can then do slightly more elaborate error checking based on this function; as seen from the C code it could have been `static`. To be visible for `dlopen()` from Python it can not be `static` - but I agree it could be removed from the header; can of course also change the name - but that is "reasonably consistent" - or at least not wildly out of line - with the rest of the code.

Reply:
> I must say that I find this API very peculiar.

Hehe - yes I can understand that; it is ~10 years old. It is documented in the `ecl_util.c` file; will add argument names in the header.